streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -32,7 +33,17 @@
  * @param <V> The value type
  */
 @InterfaceStability.Unstable
-public interface KeyValueStore<K, V> extends StateStore {
+public abstract class KeyValueStore<K, V> implements StateStore {
+
+    private boolean checked = false;
+
+    public void checkKeyIsArray(K key) {
+        if (!checked && key.getClass().isArray()) {
+            throw new StreamsException("KeyValueStore cannot support array keys.");
+        } else {
+            checked = true;
+        }
+    }
```

Comment:
@ijuma Yeah I think this is a tough call, thinking about it a bit. I feel documentation would probably be a better solution, and later if we change the serde with the class object registration, then we can just do this check once at the initialization phase, instead of on each put (though it will be triggered once).

@joshng I have once thought about moving the serde to be applied in front, but realized that could incur unnecessary serialization and deserialization processes, for change log messages produced to Kafka and rocksDB records. Hence I did not do that.